linode/lke/cluster.go

```diff
@@ -342,6 +359,20 @@ func matchPoolsWithSchema(pools []linodego.LKENodePool, declaredPools []interfac
 				continue
 			}
 
+			declaredTaints := expandNodePoolTaints(helper.ExpandObjectSet(declaredPool["taint"].(*schema.Set)))
+
+			// length comparison here is for handling the case of nil vs empty slice
+			if !reflect.DeepEqual(declaredTaints, apiPool.Taints) && len(declaredTaints) != len(apiPool.Taints) {
```

Comment:
Given order isn't guaranteed for taints, should we account for that in this comparison?

Reply:
I think order of a list of object is guaranteed by the API? Otherwise we will have a lot more semantic equality checks to implement...

Reply:
Ahh okay, the piece I was missing is that the pool is also created using `(*schema.Set).List()`, and I also did confirm that the `taints` field is effectively treated as ordered in the API 👍 

The only case I can think of that might cause issues is during an import of an LKE cluster with the same taints in a different order, although I don't think this comparison would be reached since the import would pre-populated in state.

I still think ignoring order for this comparison would be good from a resiliency perspective, but I also think keeping it as-is is a valid approach 🙂

Reply:
One more thing:

I might be misunderstanding the logic here, but it seems like using `&&` in this comparison would cause any taints list same length to pass this check regardless of their contents.

Reply:
Thanks @lgarber-akamai! I think these concerns has been addressed in the following up commits in this PR. Can you take a second look at it? Thanks!